src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,185 @@
+            preparedStatement.setLong(2, order.getOrderId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order with id "
+                    + order.getOrderId(), e);
+        }
+        deleteOrderProducts(order.getOrderId());
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement("UPDATE orders "
+                        + "SET is_deleted = TRUE WHERE order_id = ?")) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Order with id: " + id, e);
+        }
+        return deleteOrderProducts(id);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement("SELECT * FROM orders WHERE is_deleted = FALSE")) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(new Order(resultSet.getLong("order_id"),
+                        resultSet.getLong("user_id")));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add data to Orders_Product", e);
+        }
+        setProductsToOrders(orders);
+        return orders;
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders WHERE user_id = ? AND deleted = false;";
+        List<Order> orderList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long orderId = resultSet.getLong("order_id");
+                orderList.add(new Order(orderId, userId));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get orders of user with id" + userId, e);
+        }
+        for (Order order : orderList) {
+            List<Product> productList = getOrdersProducts(order.getOrderId());
+            order.setProducts(productList);
+        }
+        return orderList;
+    }
+
+    @Override
+    public Optional<Order> getByUserId(Long id) {
+        return Optional.empty();
+    }
+
+    private void addDataToOrdersProductTable(Long orderId, Long productId) {
+        try (Connection connection = ConnectionUtil.getConnection();
```

Comment:
It is best not to open a connection inside a connection. You can pass `Connection connection` as an argument to private functions where you need to create a new Statement.

Reply:
it`s used outside of try-with-recourse statement (create function) , where i using connection, so  i`m supposed either use it inside try and pass connection next, or left it as is...